announcement/models/announcement.py

```diff
@@ -70,10 +71,17 @@ def _inverse_specific_user_ids(self):
             ):
                 user.unread_announcement_ids |= announcement
 
-    @api.depends("specific_user_ids", "user_group_ids")
+    @api.depends("specific_user_ids", "user_group_ids", "is_general_announcement")
     def _compute_allowed_user_ids(self):
         self.allowed_user_ids = False
         self.allowed_users_count = False
+        # If the option "General announcement" is checked, we assign the group
+        # "Internal users" to the user_group_ids field.
+        if self.filtered(lambda x: x.is_general_announcement):
+            for announcement in self:
```

Comment:
Here you're assuming that `is_general_announcement` will be always truthy and the same for every record.

Reply:
Sorry, I don't know what I was thinking... corrected, thank you very much!